packages/actor-rdf-parse-html-script/lib/ActorRdfParseHtmlScript.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  ActorRdfParseFixedMediaTypes,
+  IActionRdfParse,
+  IActionRootRdfParse,
+  IActorOutputRootRdfParse,
+  IActorRdfParseFixedMediaTypesArgs,
+  IActorRdfParseOutput,
+  IActorTestRootRdfParse,
+} from "@comunica/bus-rdf-parse";
+import {ActionContext, Actor, Mediator} from "@comunica/core";
+import {Readable} from "stream";
+
+/**
+ * A HTML RDF Parse actor that listens on the 'rdf-parse' bus.
+ *
+ * It is able to extract JSON-LD from HTML files and parse the JSON-LD based RDF serializations
+ * and announce the presence of them by media type.
+ */
+export class ActorRdfParseHtmlScript extends ActorRdfParseFixedMediaTypes {
+
+  public mediatorRdfParse: Mediator<Actor<IActionRootRdfParse, IActorTestRootRdfParse,
+    IActorOutputRootRdfParse>, IActionRootRdfParse, IActorTestRootRdfParse, IActorOutputRootRdfParse>;
+
+  constructor(args: IActorRdfParseFixedMediaTypesArgs) {
+    super(args);
+  }
+
+  public async runHandle(action: IActionRdfParse, mediaType: string, context: ActionContext):
+    Promise<IActorRdfParseOutput> {
+
+    const myQuads = new Readable({objectMode: true});
+
+    let initialized = false;
+
+    myQuads._read = async () => {
+      if (!initialized) {
+        initialized = true;
+
+        // Stringify HTML input
+        const htmlString: string = await require('stream-to-string')(action.input);
+
+        // Extract script tags
+        const DOMParser = require('xmldom').DOMParser;
```

Comment:
It looks like `xmldom` is not aware of the browser environment yet: https://github.com/jindw/xmldom/pull/152
This means that when this is executed in the browser, that the `xmldom` code will be used, instead of the browsers's built-in DOM parser.

@pietercolpaert How do you handle this in ldfetch? Perhaps you can help out here?

Reply:
Also, `xmldom` is not present in your `package.json` file.

Reply:
Using htmlparser2 now (https://github.com/fb55/htmlparser2/).